feature_engine/datetime/datetime_subtraction.py

```diff
@@ -0,0 +1,272 @@
+
+    Methods
+    -------
+    {fit}
+
+    {fit_transform}
+
+    {transform}
+
+    """
+
+    def __init__(
+        self,
+        variables: List[Union[str, int]],
+        reference: List[Union[str, int]],
+        missing_values: str = "ignore",
+        drop_original: bool = False,
+        output_unit: str = 'D',
+    ) -> None:
+
+        if (
+            not isinstance(variables, list)
+            or not all(isinstance(var, (int, str)) for var in variables)
+            or len(set(variables)) != len(variables)
+        ):
+            raise ValueError(
+                "variables must be a list of strings or integers. "
+                f"Got {variables} instead."
+            )
+
+        if (
+            not isinstance(reference, list)
+            or not all(isinstance(var, (int, str)) for var in reference)
+            or len(set(reference)) != len(reference)
+        ):
+            raise ValueError(
+                "reference must be a list of strings or integers. "
+                f"Got {reference} instead."
+            )
+
+        if not isinstance(drop_original, bool):
+            raise ValueError(
+                "drop_original takes only booleans True or False. "
+                f"Got {drop_original} instead."
+            )
+
+        if missing_values not in ["raise", "ignore"]:
+            raise ValueError(
+                "missing_values takes only values 'raise' or 'ignore'. "
+                f"Got {missing_values} instead."
+            )
+
+        valid_output_units = {'D', 'Y', 'M', 'W', 'h', 'm', 's', 'ms', 'us', 'μs', 'ns',
+                              'ps', 'fs', 'as'}
+
+        if output_unit not in valid_output_units:
+            raise ValueError(f"output_unit accepts the following values: "
+                             f"{valid_output_units}")
+
+        self.variables = variables
+        self.reference = reference
+        self.drop_original = drop_original
+        self.missing_values = missing_values
+        self.output_unit = output_unit
+
+    def fit(self, X: pd.DataFrame, y: Optional[pd.Series] = None):
+        """
+        This transformer does not learn any parameter.
+
+        Parameters
+        ----------
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The training input samples. Can be the entire dataframe, not just the
+            variables to transform.
+
+        y: pandas Series, or np.array. Default=None.
+            It is not needed in this transformer. You can pass y or None.
+        """
+        # Common checks and attributes
+        X = check_X(X)
+
+        # check variables are datetime
+        self.reference = _find_or_check_datetime_variables(X, self.reference)
+        self.variables = _find_or_check_datetime_variables(X, self.variables)
+
+        # check if dataset contains na
+        if self.missing_values == "raise":
+            _check_contains_na(X, self.variables)
+            _check_contains_inf(X, self.variables)
```

Comment:
Does a check for inf work with datetime variables? I thought it was just with numerical. But I may be wrong.

Reply:
I don't think so. I'll remove it.